### PR TITLE
<ConceptMap /> updates - path course classification, and command updates

### PR DESCRIPTION
**Pull request by neenjaw**

### CODE REVIEW

File: app/commands/user_track/generate_concept_exercise_mapping.rb [hunk 1]
```diff
@@ -5,30 +5,43 @@ class GenerateConceptExerciseMapping
...
+          where(exercise_id:
+            Solution.where(exercise_id: user_track.track.practice_exercises).completed.select(:exercise_id)).
+          group(:track_concept_id).
+          count
+      solved_prereq_counts.default = 0
+
+      user_track.track.concepts.each_with_object({}) do |concept, hash|
+        hash[concept.slug] = {
+          exercises: prereq_counts[concept.id] + taught_counts[concept.id],
+          exercises_completed: solved_prereq_counts[concept.id] + solved_taught_counts[concept.id]
```
Comment (iHiD) [inline app/commands/user_track/generate_concept_exercise_mapping.rb]: If you use `.to_i` here, then you can remove the `default = 0` lines above. Makes it a bit more concise.

```suggestion
          exercises: prereq_counts[concept.id.to_i] + taught_counts[concept.id.to_i],
          exercises_completed: solved_prereq_counts[concept.id.to_i] + solved_taught_counts[concept.id.to_i]
```

File: app/commands/user_track/generate_concept_status_mapping.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@ class GenerateConceptStatusMapping
     initialize_with :user_track
 
     def call
+      return {} if user_track.nil?
```
Comment (iHiD) [inline app/commands/user_track/generate_concept_status_mapping.rb]: Thanks :)

Comment (iHiD) [reply app/commands/user_track/generate_concept_status_mapping.rb]: This is more idiomatic:

```suggestion
      return {} unless user_track
```

File: app/commands/user_track/generate_concept_exercise_mapping.rb (hunk 1 above)
Comment (neenjaw) [reply app/commands/user_track/generate_concept_exercise_mapping.rb]: done

File: app/commands/user_track/generate_concept_status_mapping.rb (hunk 2 above)
Comment (neenjaw) [reply app/commands/user_track/generate_concept_status_mapping.rb]: done

**Comment by iHiD** [comment]
Nice. This feels quite solid now. Next up - caching! (but that can wait for a while! :))

**Pull request merged**